Benchmarks/GroupByVersusForLoopDictionary.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Attributes;
+
+namespace Benchmarks.Benchmarks {
+	/// <summary>
+	/// Question: How long does it take to turn an enumerable into a dictionary key'd off of one of the enumerable's properties
+	/// Methods: 
+	///		ForLoopDictionary - Have a dictionary, do lookup/insert of the key in a for loop
+	///		GroupBy (Linq) - Run GroupBy, then ToDictionary linq methods
+	/// Data: All keys are ints
+	///		DistinctList - No repeated keys in list
+	///		AllSameKey - Only 1 repeated key in list
+	///		100Keys - 100 possible keys, repeated via modulo throughout the source list
+	/// Results: Forloop dictionary took about half the time of the groupby method and was slightly more memory efficient
+	///		DistinctList and hundred buckets were an order of magnitude slower than all the same key in both methods
+	/// </summary>
+	[MemoryDiagnoser]
+	public class GroupByVersusForLoopDictionary {
+		private List<(int, string)> distinctList = Enumerable.Range(0, 100_000).Select(x => (x, x.ToString())).ToList();
```

Comment:
I'd probably use a const to represent 100_000 here so that all 3 lists are ensured to always have the same # of elements. Otherwise, you may forget to change one and now your benchmarks are invalid.